addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/AbstractBulbInterface.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.milight.internal.protocol;
+
+import java.net.SocketException;
+
+import org.openhab.binding.milight.internal.MilightThingState;
+
+/**
+ * Implement this bulb interface for each new bulb type. It is used by {@see MilightLedHandler} to handle commands.
+ *
+ * @author David Graeff <david.graeff@web.de>
+ * @since 2.1
+ */
+public abstract class AbstractBulbInterface {
+    /**
+     *
+     * @param hue A value from 0 to 360
+     * @param saturation A saturation value. Can be -1 if not known
+     * @param brightness A brightness value. Can be -1 if not known
+     * @param state The changed values will be written back to the state
+     */
+    public abstract void setHSB(int hue, int saturation, int brightness, MilightThingState state);
+
+    public abstract void setPower(boolean on, MilightThingState state);
+
+    /**
+     * Switches to white mode (disables color leds).
+     *
+     * @param state The changed values will be written back to the state
+     */
+    public abstract void whiteMode(MilightThingState state);
+
+    /**
+     * Switches to night mode (low current for all leds).
+     *
+     * @param state The changed values will be written back to the state
+     */
+    public abstract void nightMode(MilightThingState state);
+
+    public abstract void setColorTemperature(int color_temp, MilightThingState state);
+
+    public abstract void changeColorTemperature(int color_temp_relative, MilightThingState state);
+
+    public abstract void setBrightness(int value, MilightThingState state);
+
+    public abstract void setSaturation(int value, MilightThingState state);
+
+    public abstract void changeBrightness(int relative_brightness, MilightThingState state);
```

Comment:
Is changeBrightness needed in this interface as it doesn't seem to be called from the handler?